### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by EmilyBourne**
Split up CI to allow for better summaries.
Add a CI to test spelling.
Clean up spelling in markdown files.
Add a CI to ensure pyccel conventions are followed (`__slots__`, `__all__` and `_attribute_nodes` in `pyccel.ast`)
Fix classes which didn't correctly follow conventions

### CODE REVIEW

File: ci_tools/summarise_pyspelling.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            print(file=f)
+            all_words.update(words)
+
+        print("These errors may be due to typos, capitalisation errors, or lack of quotes around code. If this is a false positive please add your word to `.dict_custom.txt`", file=f)
+
+        with open(os.path.join(os.path.dirname(__file__),'..','.dict_custom.txt'), encoding="utf-8") as d:
+            internal_dict = [w.strip() for w in d.readlines()]
+
+        suggestion_output = ''
+        for word in all_words:
```
Comment (jalalium) [inline ci_tools/summarise_pyspelling.py:56]: What do you think about intertwining suggestions and errors?

File: ci_tools/check_slots.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+parser = argparse.ArgumentParser(description='Check that all new lines in the python files in the pyccel/ code folder are used in the tests')
+parser.add_argument('output', metavar='output', type=str,
+                        help='File where the markdown output will be printed')
+
+args = parser.parse_args()
+
+# Get ast modules
+ast_folder = os.path.dirname(ast.__file__)
+ast_modules = [mod[:-3] for mod in os.listdir(ast_folder) if mod != '__init__.py' and mod.endswith('.py')]
```
Comment (jalalium) [inline ci_tools/check_slots.py:19]: why is the slots check restricted to the ast modules?

File: ci_tools/summarise_pyspelling.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+""" Script to output pyspelling results in a more digestible markdown format
+"""
+import argparse
+import difflib
+import os
+import sys
+
+parser = argparse.ArgumentParser(description='Check that all new lines in the python files in the pyccel/ code folder are used in the tests')
```
Comment (jalalium) [inline ci_tools/summarise_pyspelling.py:8]: The description does not match here

File: ci_tools/summarise_pyspelling.py (hunk 1 above)
Comment (EmilyBourne) [reply ci_tools/summarise_pyspelling.py]: Like this? 63482108

File: ci_tools/summarise_pyspelling.py (hunk 3 above)
Comment (EmilyBourne) [reply ci_tools/summarise_pyspelling.py]: Oops. Fixed 9351b88f

File: ci_tools/check_slots.py (hunk 2 above)
Comment (yguclu) [reply ci_tools/check_slots.py:19]: I agree with @EmilyBourne

**Review by yguclu — APPROVED** [review]

File: ci_tools/summarise_pyspelling.py (hunk 1 above)
Comment (jalalium) [reply ci_tools/summarise_pyspelling.py]: Yes thank you!

**Pull request merged**